Legacy unit tests (manager) intermittently fail with ld OOM while linking engine unit tests

# Description

The `5.X - Wazuh - Legacy Unit tests` workflow (job `Unit-Tests (manager)`) intermittently fails during the "Build wazuh" step while linking the engine unit-test binaries. The linker is terminated by the kernel OOM killer:

```
[100%] Linking CXX executable builder_utest
collect2: fatal error: ld terminated with signal 9 [Killed]
compilation terminated.
make[4]: *** [.../builder_utest] Error 1
make[2]: *** [Makefile:101: all] Error 2
make[1]: *** [Makefile:370: build_wazuh_cmake] Error 2
make: *** [Makefile:330: manager] Error 2
```

Example run: https://github.com/wazuh/wazuh/actions/runs/30029573563/job/89282566332

## RCA

The build step runs `make -C src TARGET=manager TEST=1 -j$(nproc)`. With `TEST=1` the engine unit-test executables are built. The three heaviest engine binaries — `builder_utest`, `api_tester_utest` and `wazuh-engine` — each statically link dozens of archives (abseil, RocksDB, etc.), so a single `ld` invocation peaks at several GB of RSS. Under `-j$(nproc)` their link steps overlap at the tail of the build (all reach 100% around the same time), the runner's memory is exhausted, and the OOM killer sends SIGKILL to one `ld`. Whether the peaks overlap depends on scheduling, which is why the failure is intermittent.

The engine is built with the Unix Makefiles generator, so CMake `JOB_POOLS` (which could cap concurrent link jobs) are not available. No custom linker is configured.

## DoD

- [ ] RCA
- [ ] Fix proposal

--- a/.github/actions/legacy_unit_tests.sh
+++ b/.github/actions/legacy_unit_tests.sh
@@ -14,7 +14,12 @@ build_wazuh_test_flags() {
     echo "Building Wazuh for target: $target"
     cd "$GITHUB_WORKSPACE"
     make deps -C src TARGET=${target} -j$(nproc)
-    make -C src TARGET=${target} TEST=1 -j$(nproc)
+    # Building the engine unit tests at -j$(nproc) runs many heavy compiles and links at
+    # once; their combined peak memory exhausts the runner and the OOM killer aborts the
+    # build (ld or cc1plus terminated with SIGKILL). Halve the job count to keep the peak
+    # in bounds while still using half the cores.
+    local jobs=$(( $(nproc) / 2 )); [ "$jobs" -lt 1 ] && jobs=1
+    make -C src TARGET=${target} TEST=1 -j"$jobs"
 }
 
 build_wazuh_unit_tests() {

--- a/.github/workflows/5_testunit_linux-win.yml
+++ b/.github/workflows/5_testunit_linux-win.yml
@@ -34,7 +34,9 @@ jobs:
           matrix:
               target: [agent, winagent, manager]
     runs-on: codebuild-github-actions-codebuild-runner-server-amd-${{ github.run_id }}-${{ github.run_attempt }}
-    timeout-minutes: 90
+    # Halving the build parallelism to avoid OOM (see legacy_unit_tests.sh) roughly doubles the
+    # engine unit-test compile phase, which alone approaches the old 90-minute cap.
+    timeout-minutes: 130
     steps:
       - name: Checkout Repo
         uses: actions/checkout@v5

--- a/.github/workflows/5_testunit_linux-win.yml
+++ b/.github/workflows/5_testunit_linux-win.yml
@@ -8,6 +8,7 @@ on:
         - ".github/workflows/5_testunit_linux-win.yml"
         - ".github/actions/build_test_flags/**"
         - ".github/actions/install_build_deps/**"
+        - ".github/actions/legacy_unit_tests.sh"
         - ".github/actions/legacy_unit_tests_build/**"
         - ".github/actions/legacy_unit_tests_run/**"
         - ".github/actions/reinstall_cmake/**"

